src/app/api/auth/[...nextauth]/authOptions.ts

```diff
@@ -9,152 +9,31 @@
-
-                    const response = await ApiUtils.GET(`users/${username}`, authToken)
-                    if (response.status !== HttpStatus.OK) {
-                        throw new Error('Error while fetching User Group')
-                    }
-                    const data = await response.json()
-                    return { access_token: authToken, userGroup: data.userGroup, email: username} as any
-                } catch (e) {
-                    console.error(e)
-                    return null
-                }
-            },
-        }),
-        CredentialsProvider({
-            name: CREDENTIAL_PROVIDER,
-            credentials: {},
-            async authorize(credentials) {
-                try {
-                    const { username, password } = credentials as any
-                    const userCredential: UserCredentialInfo = {
-                        username: username,
-                        password: password,
-                    }
-
-                    const authToken = await AuthService.generateToken(userCredential)
-
-                    if (authToken === null) throw new Error('Error while fetching Auth Token')
-                    
-                    const response = await ApiUtils.GET(`users/${username}`, authToken.access_token)
-                    if (response.status !== HttpStatus.OK) {
-                        throw new Error('Error while fetching User Group')
-                    }
-                    
-                    const data = await response.json()
-                    return { ...authToken, userGroup: data.userGroup, email: username } as any
-                } catch (e) {
-                    console.error(e)
-                    return null
-                }
-            },
-        }),
-    ],
-
-    session: {
-        strategy: 'jwt',
-    },
-
-    callbacks: {
-        async jwt({ token, user }) {
-            return { ...token, ...user }
-        },
-        async session({ session, token }) {
-            // Send properties to the client, like an access_token from a provider.
-            session.user = token
-            return session
-        },
-    },
-
-    pages: {
-        signIn: '/',
-    },
+import { AUTH_PROVIDER } from '@/utils/env';
+import keycloakAuth from './keycloakAuthOption';
+import { NextAuthOptions } from 'next-auth';
+import basicAuthOption from './basicAuthOption';
+import sw360OauthOption from './sw360OauthOption';
+import sw360OauthPwdGrantTypeOption from './sw360OauthPwdGrantTypeOption';
+
+let authOptions: NextAuthOptions
+
+switch (AUTH_PROVIDER) {
+    case 'sw360basic':
+        authOptions = basicAuthOption
+        break;
+    case 'sw360oauth':
+        authOptions = sw360OauthOption
+        break;
+    case 'keycloak':
+        authOptions = keycloakAuth
+        break;
+    case 'oauth-password-grant':
+        authOptions = sw360OauthPwdGrantTypeOption
+        break;
+    default:
+        break;
```

Comment:
I think we should set basicAuthOption in default case

Reply:
Done